### PR TITLE
Fix pagination #5605

### LINKED ISSUE #5605: Fix pagination
- should paginate to 100 instead of 12
- locations route pagination has strange bug, second page is empty
- users table sort completely broken
- rather than sorting on the client side, the table should populate from the feathers database query

### PR DESCRIPTION
**Pull request by kinjalravel**
## Summary

Change page size, fixed second page empty issue, fixed users table sort issue and sort users and location data from server

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

closes #5605

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

**Comment by zulqarnainhanif** [comment]
@kinjalravel as far as I understood the 4th point of the issue; We need sorting on multiple fields and every table of admin panel. Currently it's only done for name field of User and Location Table.

@HexaField Please let us know if my understanding is correct or not.

**Comment by kinjalravel** [comment]
> @kinjalravel as far as I understood the 4th point of the issue; We need sorting on multiple fields and every table of admin panel. Currently it's only done for name field of User and Location Table.
> 
> @HexaField Please let us know if my understanding is correct or not.

@HexaField ?

**Comment by HexaField** [comment]
> > @kinjalravel as far as I understood the 4th point of the issue; We need sorting on multiple fields and every table of admin panel. Currently it's only done for name field of User and Location Table.
> > @HexaField Please let us know if my understanding is correct or not.
> 
> @HexaField ?

no, this is referring to the fact that at least one of the tables is sorting the data it has been given from the server, rather than sending the feathers database query information for the pagination

File: packages/client-core/src/admin/common/Table.tsx [hunk 1]
```diff
@@ -19,6 +19,10 @@ interface Props {
   page: number
   rowsPerPage: number
   count: number
+  orderby?: string
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/Table.tsx:22]: proper camel case would be `orderBy`

File: packages/client-core/src/admin/common/Table.tsx [hunk 2]
```diff
@@ -19,6 +19,10 @@ interface Props {
...
+  allowSort?: boolean
+  setSortField?: (fueld: string) => void
+  setOrderby?: (order: string) => void
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/Table.tsx:25]: proper camel case would be `setOrderBy`

File: packages/client-core/src/admin/common/Table.tsx [hunk 3]
```diff
@@ -143,7 +161,7 @@ const TableComponent = (props: Props) => {
             columns={column}
           />
           <TableBody>
-            {stableSort(rows, getComparator(order, orderBy))
+            {(allowSort === false ? rows : stableSort(rows, getComparator(order, orderBy)))
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/Table.tsx:164]: `allowSort ? stableSort(rows, getComparator(order, orderBy) : rows`

File: packages/client-core/src/admin/common/hooks/Location.hooks.ts [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
+export const useFetchLocation = (user, adminLocationState, adminScopeReadErrMsg, search, LocationService, orderby) => {
   useEffect(() => {
-    if (user?.id?.value && adminLocationState.updateNeeded.value && !adminScopeReadErrMsg?.value) {
-      LocationService.fetchAdminLocations('increment', null)
-    }
-    // if (search) {
-    LocationService.fetchAdminLocations('increment', search)
-    // }
+    //if (user?.id?.value && adminLocationState.updateNeeded.value && !adminScopeReadErrMsg?.value) {
+    //  LocationService.fetchAdminLocations(search, 0, orderby)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/Location.hooks.ts:6]: remove commented code if it isn't required anymore.

File: packages/client-core/src/admin/common/hooks/User.hooks.ts [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 
-export const useFetchUsersAsAdmin = (user, adminUserState, UserService, search) => {
+export const useFetchUsersAsAdmin = (user, adminUserState, UserService, search, sortField, orderby) => {
   useEffect(() => {
-    if (user?.id.value && adminUserState.updateNeeded.value) {
-      UserService.fetchUsersAsAdmin('increment', null)
-    }
-    UserService.fetchUsersAsAdmin('increment', search)
+    //if (user?.id.value && adminUserState.updateNeeded.value) {
+    // UserService.fetchUsersAsAdmin(search, 0, orderby)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/User.hooks.ts:6]: remove commented code if it isn't required anymore.

File: packages/client-core/src/admin/common/hooks/party.hooks.ts [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
 
-export const useFetchAdminParty = (user, adminParty, adminPartyState, PartyService, search) => {
+export const useFetchAdminParty = (user, adminPartyState, PartyService, search, page, sortField, orderby) => {
   useEffect(() => {
-    if (user?.id?.value && adminParty.updateNeeded.value) {
-      PartyService.fetchAdminParty('increment', null)
-    }
-    PartyService.fetchAdminParty('increment', search)
+    //if (user?.id?.value && adminParty.updateNeeded.value) {
+    //PartyService.fetchAdminParty('increment', null)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/party.hooks.ts:6]: remove commented code if it isn't required anymore.

File: packages/client-core/src/admin/components/Avatars/AvatarTable.tsx [hunk 7]
```diff
@@ -36,15 +36,23 @@ const AvatarTable = (props: Props) => {
   const [popConfirmOpen, setPopConfirmOpen] = useState(false)
   const [avatarId, setAvatarId] = useState('')
   const [avatarName, setAvatarName] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Avatars/AvatarTable.tsx:39]: fix camelcase here as well

File: packages/client-core/src/admin/components/Group/GroupTable.tsx [hunk 8]
```diff
@@ -24,18 +24,26 @@ const GroupTable = (props: Props) => {
   const [rowsPerPage, setRowsPerPage] = useState(GROUP_PAGE_LIMIT)
   const [groupId, setGroupId] = useState('')
   const [groupName, setGroupName] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Group/GroupTable.tsx:27]: fix camelcase here as well

File: packages/client-core/src/admin/components/Instance/InstanceTable.tsx [hunk 9]
```diff
@@ -25,24 +25,31 @@ interface Props {
...
-  const [popConfirmOpen, setPopConfirmOpen] = React.useState(false)
-  const [instanceId, setInstanceId] = React.useState('')
-  const [instanceName, setInstanceName] = React.useState('')
+  const [page, setPage] = useState(0)
+  const [rowsPerPage, setRowsPerPage] = useState(INSTNCE_PAGE_LIMIT)
+  const [refetch, setRefetch] = useState(false)
+  const [popConfirmOpen, setPopConfirmOpen] = useState(false)
+  const [instanceId, setInstanceId] = useState('')
+  const [instanceName, setInstanceName] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/InstanceTable.tsx:34]: fix camelcase here as well

File: packages/client-core/src/admin/components/Users/UserTable.tsx [hunk 10]
```diff
@@ -19,6 +19,8 @@ const UserTable = (props: UserProps) => {
   const [popConfirmOpen, setPopConfirmOpen] = useState(false)
   const [userId, setUserId] = useState('')
   const [userName, setUserName] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Users/UserTable.tsx:22]: fix camelcase here as well

File: packages/client-core/src/admin/components/Party/PartyTable.tsx [hunk 11]
```diff
@@ -20,6 +20,8 @@ const PartyTable = (props: PartyPropsTable) => {
   const [popConfirmOpen, setPopConfirmOpen] = useState(false)
   const [partyName, setPartyName] = useState('')
   const [partyId, setPartyId] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Party/PartyTable.tsx:23]: fix camelcase here as well

File: packages/client-core/src/admin/components/Location/LocationTable.tsx [hunk 12]
```diff
@@ -25,13 +25,15 @@ const LocationTable = (props: LocationProps) => {
...
-  const [locationId, setLocationId] = React.useState('')
-  const [locationName, setLocationName] = React.useState('')
-  const [viewModal, setViewModal] = React.useState(false)
-  const [locationAdmin, setLocationAdmin] = React.useState<Location>()
+  const [page, setPage] = useState(0)
+  const [rowsPerPage, setRowsPerPage] = useState(LOCATION_PAGE_LIMIT)
+  const [popConfirmOpen, setPopConfirmOpen] = useState(false)
+  const [locationId, setLocationId] = useState('')
+  const [locationName, setLocationName] = useState('')
+  const [orderby, setOrderby] = useState('asc')
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Location/LocationTable.tsx:33]: fix camelcase here as well

**Comment by zulqarnainhanif** [comment]
getting this error on Sorting Location Table Type column

Screenshot 2022-04-06 at 1 34 25 PM

**Review by zulqarnainhanif — APPROVED** [review]

**Pull request merged**